### DISCUSSION TITLE
Hardware revision is not supported - Mango Pi

### DISCUSSION THREAD

**Discussion by DaddyW123** [hyperion-project/hyperion.ng#1574] [Q&A]
I'm attempting to try to get Hyperion Project working on a Mango Pi MQ Quad. I already have this project working on an old Pi Zero (first gen), but there is a bit of a delay when the screen changes and the LED color change.
I was hoping to use a Pi Zero 2, but of course Pi's are hard to get a hold of these days. So I found this Mango Pi MQ Quad reviewed as a great alternative to the Pi Zero 2 and is actually even more powerful.
Anyway, fast forward - The Mango Pi can't boot using the HyperBian image. So I have to use the vendor provided Debian image (which is actually orange pi bullseye), then run the installer instructions for debian.
Everything installs just fine, other than the known issue that it doesn't create a service file in systemd folder. So I had to create that, and yes I have the service running as root as it shows when I do top | grep hyperion
So I was getting all of my settings to match my existing Pi Zero and I noticed in the Log Output window, I see the following messages:

2023-02-19T16:58:15.503Z [LEDDEVICE|GPIO-18] (ERROR) Device disabled, device 'ws281x' signals error: 'Failed to open. Error message: Hardware revision is not supported'
2023-02-19T16:59:45.585Z [LEDDEVICE|GPIO-18] (ERROR) Previous line repeats 6 times
2023-02-19T16:59:45.587Z [LEDDEVICE|GPIO-18] (ERROR) Device disabled. Maximum number of 6 attempts enabling the device reached. Tried for 105 seconds.

I'm wondering if this is because the hyperion project is built around an actual Raspberry Pi and the software doesn't know how to interface with the Mango Pi's IO? Is that possible? Or could it be something else? I also don't have an actual LED strip connected to the board at the time since I'm still setting it up, so could it be that it needs that connection present?
I'm hoping this is an easy fix because this board was only $38 from aliexpress with a nice metal heat sync case. So cost wise, it seems to be a great alternative... as long as I can get it to work :)
Any help would be much appreciated.

**Comment by Lord-Grey** [comment]
Hyperion makes use of an external [ws281x library](https://github.com/jgarff/rpi_ws281x/tree/1ba8e385708fb7802b09c0177a7ea4293948e25c) 
The error "Hardware revision is not supported" comes from the given library.
Maybe you ask there, if you specific board could/will be supported.

Nevertheless, you can make still use of the other available features of Hyperion.
You might want exploring to make use of an ESP running WLED where your ws281x LED are connected.

I would suggest you check with the [Hyperion Forum](https://hyperion-project.org/forum/), as there are various ways how people run Hyperion out there....

**Reply by DaddyW123** [reply]
Well that makes sense. Bummer. I do have a bunch of spare ESP8266's for wled because they come in a pack.  But I liked the idea of my LED's being directly connected for simplicity sake. 
Do they perform just as well as directly connected? I would think the latency of sending the command via wifi would cause a delay in color change. 
Plus if your wifi goes down, no lights.

**Reply by tekdroid** [reply]
Not sure if this is helpful info or not, but I ran in to same issue with a Le Potato and ended up using a usbasp device (bought a 2 pack on amazon, one is to flash the other. I also had to jump the pins myself as it didn't come with a jumper.) and it works really well. The LED strip I got even came with a connector I'm using to connect to the 6 port connector. I wired the DI and Ground wires to it, then wired ground and live to the 5v power supply. Wiring ground to the usbasp took me several hours to figure out, the LEDs were not operating properly before doing so. I only found hints of this in diagrams found via Google. 

 I hope this might help, even a little.

**Comment by jnnutt** [comment]
Hi @DaddyW123 ,
have you been able to fix the problem in any way? I am currently facing a similiar situation using a Banana Pi BPI-M2 ZERO.
Did you for example manage to get it running with an extra esp?

**Reply by DaddyW123** [reply]
@jnnutt, Sorry for the delayed response.  I actually put this project on the back burner.  I did pull out one of my spare ESP8266's and loaded WLED on it, and even connected the Hyperion install to the WLED in the interface.  But that's where I left it.  I never actually got around to physically hooking up any LED's to the WLED board.  Nor did I get a video capture device hooked up to the Mango Pi.  Funny thing is, both devices are sitting on my workbench in my basement still powered on.  I just moved on to other things.  I'm sure I'll revisit this someday but I have so many home hobbies that I tend to bounce around a little too much and a lot of things just end up never getting done.
